File: problem1.java
long oddNumbers(String s)
{
    long choices = 1;
    int possible = 9;
    ArrayList<Character> used = new ArrayList <Character>();
    for(int i =0; i < s.length(); i ++){
        char a = s.charAt(s.length()- i - 1);
        if(i == 0){
            choices *= 5;
            used.add(s.charAt(s.length()- i -1));
        }
        else if(used.contains(a)){

            choices *=1;
        }
        else{
            choices *= possible;
            possible -=1;
            used.add(a);
        }
    }
    return choices;
}

File: problem1.py
def oddNumbers(s):
    choices = 1
    possible = 9
    used = []
    for i in range(len(s)):
        if i == 0:
            choices *= 5
        elif s[-i -1] in used:
            choices *= 1
        else:
            choices *= possible
            possible -= 1
        used.append(s[-i -1])
    return choices


print(oddNumbers("bbbbbbbbbbbbbbbccccccbbbbbbeeezzzxxx"))
